### PR TITLE
/ban and /howgay

### PR DESCRIPTION
**Pull request by Hades3235**
guess it can be useful for new gtps's people make made by @niko1802

### CODE REVIEW

**Comment by ardakurucim** [comment]
i think you should add them into /help

File: enet server test/enet server test.cpp [hunk 1]
```diff
@@ -2738,6 +2738,24 @@ int _tmain(int argc, _TCHAR* argv[])
 						delete p.data;
 						//enet_host_flush(server);
 					}
+                   else if (str.substr(0, 8) == "/howgay") {
```
Comment (GrowtopiaNoobs) [inline enet server test/enet server test.cpp]: I haven't checked this, but I think that this is wrong.
```suggestion
                   else if (str.substr(0, 7) == "/howgay") {
```

File: enet server test/enet server test.cpp [hunk 2]
```diff
@@ -2592,6 +2592,64 @@ int _tmain(int argc, _TCHAR* argv[])
 							ENET_PACKET_FLAG_RELIABLE);
 						enet_peer_send(peer, 0, packet);
 					}
+                                        else if (str.substr(0, 5) == "/ban ") {
+						if (getAdminLevel(((PlayerInfo*)(peer->data))->rawName, ((PlayerInfo*)(peer->data))->tankIDPass) > 0) {
+							string name = str.substr(5, str.length());
```
Comment (GrowtopiaNoobs) [inline enet server test/enet server test.cpp]: I think this is wrong too, it wont be whole string length, but I think length()-6 (test it).

File: enet server test/enet server test.cpp [hunk 3]
```diff
@@ -2592,6 +2592,64 @@ int _tmain(int argc, _TCHAR* argv[])
...
+									continue;
+
+								if (((PlayerInfo*)(currentPeer->data))->rawName == name) {
+									found = true;
+									std::ifstream ifs("players/" + ((PlayerInfo*)(currentPeer->data))->rawName + ".json");
+									if (ifs.is_open()) {
+										json x;
+										ifs >> x;
+										std::ofstream o("players/" + ((PlayerInfo*)(currentPeer->data))->rawName + ".json");
+										json j;
```
Comment (GrowtopiaNoobs) [inline enet server test/enet server test.cpp]: You don't need to copy content of those json fields one by one, but you can just use that existing json object `x` and just add field `"banned"` and then save it.

**Comment by GrowtopiaNoobs** [comment]
Those are all issues I have seen. Good job :smiley: and thank you for your work. If you fix those, then I will merge it.

(Btw you might add check if player is banned on logging and as @ardakurucim suggested, you can add them to /help, or for /ban you can create /ahelp.)

**Comment by Yunus971** [comment]
I'm banning from game but he can join more. I did all.

**Pull request closed without merging**